pkgs/by-name/ra/radiance/package.nix

```
@@ -0,0 +1,34 @@
+{
+  lib,
+  stdenv,
+  cmake,
+  libGL,
+  libGLU,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "radiance";
+  version = "5.4";
+
+  src = fetchFromGitHub {
+    owner = "LBNL-ETA";
+    repo = pname;
+    rev = "3be7601c";
```

According to the upstream releases version "5.4" is tagged as [`rad5R4`](https://github.com/LBNL-ETA/Radiance/releases/tag/rad5R4) pointing to commit [`4ee32974`](https://github.com/LBNL-ETA/Radiance/commit/4ee32974b15851bd7f2cfc45153d5ddc9a75f324) released on 2023-11-05
The commit [`3be7601c`](https://github.com/LBNL-ETA/Radiance/commit/3be7601c) used in this package was created 2 weeks ago. What was the motivation to use that commit rather than the official release tag?

```suggestion
    rev = "rad5R4";
```

---

Thanks for the catch. Not sure where I even got that commit from 😱

---

Oof, this version doesn't build. The newer 6.0 pre release versions do build successfully. Is it legit to use one of these?

---

I don't really have any clue on how to fix v5.4. If you want to give it a look I'm posting the stderr output here. Otherwise I'd just roll with the 6.0a release. I'll ask upstream for a release.

<details>
<summary> Posting compile errors here </summary>

```
this derivation will be built:
  /nix/store/jcynxhlb76yyjx3rp4g5zfnxz4sxb0k6-radiance-5.4.drv
building '/nix/store/jcynxhlb76yyjx3rp4g5zfnxz4sxb0k6-radiance-5.4.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/0wj1if0d6aawqf3kpxb82inmlhdxc85c-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/share/doc/radiance -DCMAKE_INSTALL_INFODIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/include -DCMAKE_INSTALL_SBINDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/4kwk49k8rc8cg8c387kbbjcrl8i0y693-gcc-wrapper-13.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/4kwk49k8rc8cg8c387kbbjcrl8i0y693-gcc-wrapper-13.2.0/bin/ranlib -DCMAKE_AR=/nix/store/4kwk49k8rc8cg8c387kbbjcrl8i0y693-gcc-wrapper-13.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/gjfpzbmpvkgf2rsizqgcyrkbcvs7q1dc-radiance-5.4  
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/4kwk49k8rc8cg8c387kbbjcrl8i0y693-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/4kwk49k8rc8cg8c387kbbjcrl8i0y693-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
-- Could NOT find X11 (missing: X11_X11_INCLUDE_PATH X11_X11_LIB) 
-- Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR) 
-- Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR) 
CMake Warning (dev) at /nix/store/0xpv46mb188z226j7cs211zmmi5xw9nj-cmake-3.29.1/share/cmake-3.29/Modules/FindOpenGL.cmake:415 (message):
  Policy CMP0072 is not set: FindOpenGL prefers GLVND by default when
  available.  Run "cmake --help-policy CMP0072" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  FindOpenGL found both a legacy GL library:

    OPENGL_gl_LIBRARY: /nix/store/qady90492hsb01i96amgb8z6q24bv9kd-libGL-1.7.0/lib/libGL.so

  and GLVND libraries for OpenGL and GLX:

    OPENGL_opengl_LIBRARY: /nix/store/qady90492hsb01i96amgb8z6q24bv9kd-libGL-1.7.0/lib/libOpenGL.so
    OPENGL_glx_LIBRARY: /nix/store/qady90492hsb01i96amgb8z6q24bv9kd-libGL-1.7.0/lib/libGLX.so

  OpenGL_GL_PREFERENCE has not been set to "GLVND" or "LEGACY", so for
  compatibility with CMake 3.10 and below the legacy GL library will be used.
Call Stack (most recent call first):
  src/util/CMakeLists.txt:108 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found OpenGL: /nix/store/qady90492hsb01i96amgb8z6q24bv9kd-libGL-1.7.0/lib/libOpenGL.so
-- Could NOT find Perl (missing: PERL_EXECUTABLE) 
-- Configuring done (0.4s)
-- Generating done (0.1s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
build flags: -j32 SHELL=/nix/store/dy2z01kpnxn7dn2kgfdxs4fm8xy9mb89-bash-5.2p26/bin/bash
[  0%] Generating Version.c
/nix/store/dy2z01kpnxn7dn2kgfdxs4fm8xy9mb89-bash-5.2p26/bin/bash: line 1: csh: command not found
make[2]: *** [src/util/CMakeFiles/trad.dir/build.make:70: src/util/CMakeFiles/trad] Error 127
make[1]: *** [CMakeFiles/Makefile2:4114: src/util/CMakeFiles/trad.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  1%] Building C object src/cv/CMakeFiles/nff2rad.dir/nff2rad.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/g3vector.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/g3flist.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/g3affine.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/g3sphere.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/muc_randvar.c.o
[  1%] Building C object src/common/CMakeFiles/reetz.dir/gbasic.c.o
[  1%] Building C object src/cal/CMakeFiles/histo.dir/histo.c.o
[  1%] Building C object src/common/CMakeFiles/mgf.dir/mgf_context.c.o
[  1%] Building C object src/common/CMakeFiles/mgf.dir/mgf_object.c.o
[  1%] Building C object src/meta/CMakeFiles/meta.dir/metacalls.c.o
[  1%] Building C object src/common/CMakeFiles/mgf.dir/mgf_parser.c.o
[  1%] Building C object src/common/CMakeFiles/mgf.dir/mgf_xf.c.o
[  1%] Building C object src/gen/CMakeFiles/genmarble.dir/genmarble.c.o
[  1%] Building C object src/meta/CMakeFiles/meta.dir/progname.c.o
[  1%] Building C object src/meta/CMakeFiles/meta.dir/mfio.c.o
[  1%] Building C object src/common/CMakeFiles/wfobj.dir/convertobj.c.o
[  1%] Building C object src/cal/CMakeFiles/neaten.dir/neat.c.o
[  1%] Building C object src/common/CMakeFiles/wfobj.dir/objutil.c.o
[  1%] Building C object src/meta/CMakeFiles/meta.dir/primout.c.o
[  1%] Building C object src/meta/CMakeFiles/meta.dir/syscalls.c.o
[  1%] Building C object src/cv/CMakeFiles/epw2wea.dir/epw2wea.c.o
[  2%] Building C object src/common/CMakeFiles/wfobj.dir/readwfobj.c.o
[  2%] Building C object src/meta/CMakeFiles/meta.dir/misc.c.o
[  2%] Building C object src/common/CMakeFiles/rgl.dir/rgldomat.c.o
[  2%] Building C object src/gen/CMakeFiles/genbeads.dir/genbeads.c.o
[  2%] Building C object src/gen/CMakeFiles/genclock.dir/genclock.c.o
[  2%] Building C object src/gen/CMakeFiles/genbranch.dir/genbranch.c.o
[  2%] Building C object src/common/CMakeFiles/rtrad.dir/addobjnotify.c.o
RADIANCE 5.4
[  2%] Building C object src/rt/CMakeFiles/raycalls.dir/raycalls.c.o
[  2%] Building C object src/common/CMakeFiles/rgl.dir/rglfile.c.o
[  2%] Building C object src/rt/CMakeFiles/radiance.dir/Version.c.o
[  2%] Building C object src/rt/CMakeFiles/raycalls.dir/raypcalls.c.o
/build/source/src/cv/epw2wea.c: In function 'main':
/build/source/src/cv/epw2wea.c:54:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   54 |         fscanf(EPW_FILE,"%[^,]s",keyword);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:56:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   56 |         fscanf(EPW_FILE,",%[^,]s",city);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:57:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   57 |         fscanf(EPW_FILE,",%[^,]s",country);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:58:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   58 |         fscanf(EPW_FILE,",%[^,]s",country);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:63:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   63 |         fscanf(EPW_FILE,",%[^,]s",country);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:64:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   64 |         fscanf(EPW_FILE,",%[^,]s",country);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:65:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   65 |         fscanf(EPW_FILE,",%[^,]s",latitude);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:68:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   68 |         fscanf(EPW_FILE,",%[^,]s",longitude);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:72:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   72 |         fscanf(EPW_FILE,",%[^,]s",time_zone);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:75:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   75 |         fscanf(EPW_FILE,",%s[^\n]",elevation);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:79:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   79 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:79:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   79 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:80:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   80 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:80:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   80 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:81:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   81 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:81:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   81 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:82:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   82 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:82:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   82 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:83:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   83 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:83:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   83 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:84:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   84 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:84:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   84 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:85:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   85 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:85:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   85 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:88:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   88 |         fscanf(EPW_FILE,"%[^,]s",keyword);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:89:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   89 |         fscanf(EPW_FILE,",%[^,]s",keyword);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:90:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   90 |         fscanf(EPW_FILE,",%[^,]s",minute_string);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:94:9: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   94 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:94:36: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   94 |         fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:100:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  100 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:101:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  101 |                 fscanf(EPW_FILE,",%[^,]s",city);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:102:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  102 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:103:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  103 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:104:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  104 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:105:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  105 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:106:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  106 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:107:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  107 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:108:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  108 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:109:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  109 |                 fscanf(EPW_FILE,",%f",&dummy_float);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:111:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  111 |                 fscanf(EPW_FILE,",%f,%f",&dir_norm_rad, &dif_or_rad);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:114:17: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  114 |                 fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/epw2wea.c:114:44: warning: ignoring return value of 'fscanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  114 |                 fscanf(EPW_FILE,"%*[^\n]");fscanf(EPW_FILE,"%*[\n\r]");
      |                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[  3%] Building C object src/common/CMakeFiles/rtrad.dir/badarg.c.o
[  3%] Building C object src/rt/CMakeFiles/radiance.dir/__/common/paths.c.o
/build/source/src/cv/nff2rad.c: In function 'view':
/build/source/src/cv/nff2rad.c:281:9: warning: ignoring return value of 'scanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  281 |         scanf(" hither %*f");
      |         ^~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/nff2rad.c:282:9: warning: ignoring return value of 'scanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  282 |         scanf(" resolution %*d %*d");
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/cv/nff2rad.c: In function 'poly':
/build/source/src/cv/nff2rad.c:584:25: warning: ignoring return value of 'scanf' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  584 |                         scanf("%*f %*f %*f");
      |                         ^~~~~~~~~~~~~~~~~~~~
[  3%] Building C object src/rt/CMakeFiles/raycalls.dir/rayfifo.c.o
[  3%] Building C object src/common/CMakeFiles/rtrad.dir/biggerlib.c.o
[  3%] Linking C executable ../../bin/genmarble
[  3%] Building C object src/gen/CMakeFiles/genbeads.dir/hermite3.c.o
[  4%] Building C object src/common/CMakeFiles/rgl.dir/rglinst.c.o
[  4%] Building C object src/common/CMakeFiles/rtrad.dir/bmalloc.c.o
[  4%] Linking C executable ../../bin/neaten
[  4%] Building C object src/common/CMakeFiles/rtrad.dir/bmpfile.c.o
[  4%] Building C object src/common/CMakeFiles/rgl.dir/rglmat.c.o
[  4%] Building C object src/common/CMakeFiles/rtrad.dir/bsdf.c.o
/build/source/src/common/rglfile.c: In function 'rgl_load':
/build/source/src/common/rglfile.c:175:25: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  175 |                         fgets(buf, sizeof(buf), infp);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  5%] Linking C executable ../../bin/epw2wea
[  5%] Linking C executable ../../bin/genclock
[  5%] Building C object src/common/CMakeFiles/rgl.dir/rglsrc.c.o
[  5%] Building C object src/common/CMakeFiles/rtrad.dir/bsdf_m.c.o
[  6%] Building C object src/common/CMakeFiles/rtrad.dir/bsdf_t.c.o
[  6%] Linking C executable ../../bin/genbranch
[  6%] Building C object src/common/CMakeFiles/rtrad.dir/byteswap.c.o
[  6%] Building C object src/rt/CMakeFiles/radiance.dir/ambcomp.c.o
[  6%] Linking C executable ../../bin/genbeads
[  6%] Building C object src/rt/CMakeFiles/radiance.dir/ambient.c.o
[  6%] Built target genmarble
[  6%] Linking C static library libmeta.a
[  6%] Linking C executable ../../bin/histo
[  7%] Building C object src/rt/CMakeFiles/radiance.dir/ambio.c.o
[  7%] Building C object src/common/CMakeFiles/rtrad.dir/caldefn.c.o
[  8%] Building C object src/common/CMakeFiles/wfobj.dir/writewfobj.c.o
[  8%] Built target neaten
/build/source/src/common/bsdf.c: In function 'SDloadFile':
[  8%] Building C object src/common/CMakeFiles/rgl.dir/rglsurf.c.o
/build/source/src/common/bsdf.c:188:13: warning: argument 1 null where non-null expected [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wnonnull-Wnonnull8;;]
  188 |         if (strcmp(ezxml_name(fl), "WindowElement")) {
      |             ^~~~~~
In file included from /build/source/src/common/bsdf.c:16:
/nix/store/ij144ma6vs8acil8r9hgr8xkb1dp9azg-glibc-2.39-5-dev/include/string.h:156:12: note: in a call to function 'strcmp' declared 'nonnull'
  156 | extern int strcmp (const char *__s1, const char *__s2)
      |            ^~~~~~
[  8%] Building C object src/common/CMakeFiles/wfobj.dir/objtriangulate.c.o
[  8%] Building C object src/rt/CMakeFiles/radiance.dir/aniso.c.o
[  8%] Building C object src/rt/CMakeFiles/radiance.dir/ashikhmin.c.o
[  8%] Built target epw2wea
[  8%] Building C object src/rt/CMakeFiles/radiance.dir/data.c.o
[  8%] Built target meta
[  8%] Building C object src/common/CMakeFiles/rtrad.dir/calexpr.c.o
[  8%] Building C object src/rt/CMakeFiles/radiance.dir/dielectric.c.o
/build/source/src/common/rglinst.c: In function 'checkoct':
/build/source/src/common/rglinst.c:182:13: warning: 'checkheader' accessing 64 bytes in a region of size 16 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  182 |         if (checkheader(infp, OCTFMT, NULL) < 0)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/common/rglinst.c:182:13: note: referencing argument 2 of type 'char[64]'
In file included from /build/source/src/common/standard.h:10,
                 from /build/source/src/common/radogl.h:8,
                 from /build/source/src/common/rglinst.c:17:
/build/source/src/common/rtio.h:64:17: note: in a call to function 'checkheader'
   64 | extern int      checkheader(FILE *fin, char fmt[MAXFMTLEN], FILE *fout);
      |                 ^~~~~~~~~~~
[  8%] Building C object src/common/CMakeFiles/rtrad.dir/calfunc.c.o
[  8%] Building C object src/common/CMakeFiles/rtrad.dir/calprnt.c.o
[  9%] Linking C executable ../../bin/nff2rad
[  9%] Built target genclock
[  9%] Building C object src/rt/CMakeFiles/radiance.dir/fprism.c.o
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/freeobjmem.c.o
[ 10%] Built target genbeads
[ 10%] Built target histo
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/func.c.o
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/glass.c.o
/build/source/src/rt/ambient.c: In function 'setambient':
/build/source/src/rt/ambient.c:161:25: warning: ignoring return value of 'ftruncate' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
  161 |                         ftruncate(fileno(ambfp), (off_t)lastpos);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/rt/ambient.c: In function 'initambfile':
/build/source/src/rt/ambient.c:654:20: warning: 'checkheader' accessing 64 bytes in a region of size 16 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  654 |         } else if (checkheader(ambfp, AMBFMT, NULL) < 0 || !hasambmagic(ambfp))
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/rt/ambient.c:654:20: note: referencing argument 2 of type 'char[64]'
In file included from /build/source/src/common/standard.h:10,
                 from /build/source/src/rt/ray.h:8,
                 from /build/source/src/rt/ambient.c:13:
/build/source/src/common/rtio.h:64:17: note: in a call to function 'checkheader'
   64 | extern int      checkheader(FILE *fin, char fmt[MAXFMTLEN], FILE *fout);
      |                 ^~~~~~~~~~~
[ 10%] Built target genbranch
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/initotypes.c.o
/build/source/src/common/rglsurf.c: In function 'newtess':
/build/source/src/common/rglsurf.c:115:48: warning: passing argument 3 of 'gluTessCallback' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
  115 |         gluTessCallback(gluto, GLU_TESS_BEGIN, glBegin);
      |                                                ^~~~~~~
      |                                                |
      |                                                void (*)(GLenum) {aka void (*)(unsigned int)}
In file included from /build/source/src/common/radogl.h:15,
                 from /build/source/src/common/rglsurf.c:10:
/nix/store/n3d5cj7sy7nmbhnkccalkqskj92p26kc-glu-9.0.3-dev/include/GL/glu.h:336:87: note: expected '_GLUfuncptr' {aka 'void (*)(void)'} but argument is of type 'void (*)(GLenum)' {aka 'void (*)(unsigned int)'}
  336 | GLAPI void GLAPIENTRY gluTessCallback (GLUtesselator* tess, GLenum which, _GLUfuncptr CallBackFunc);
      |                                                                           ~~~~~~~~~~~~^~~~~~~~~~~~
/build/source/src/common/rglsurf.c:116:49: warning: passing argument 3 of 'gluTessCallback' from incompatible pointer type [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
  116 |         gluTessCallback(gluto, GLU_TESS_VERTEX, glVertex3dv);
      |                                                 ^~~~~~~~~~~
      |                                                 |
      |                                                 void (*)(const GLdouble *) {aka void (*)(const double *)}
/nix/store/n3d5cj7sy7nmbhnkccalkqskj92p26kc-glu-9.0.3-dev/include/GL/glu.h:336:87: note: expected '_GLUfuncptr' {aka 'void (*)(void)'} but argument is of type 'void (*)(const GLdouble *)' {aka 'void (*)(const double *)'}
  336 | GLAPI void GLAPIENTRY gluTessCallback (GLUtesselator* tess, GLenum which, _GLUfuncptr CallBackFunc);
      |                                                                           ~~~~~~~~~~~~^~~~~~~~~~~~
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/m_alias.c.o
[ 10%] Built target nff2rad
[ 10%] Building C object src/rt/CMakeFiles/radiance.dir/m_brdf.c.o
[ 11%] Building C object src/rt/CMakeFiles/radiance.dir/m_bsdf.c.o
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/ccolor.c.o
[ 12%] Linking C static library libraycalls.a
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/ccyrgb.c.o
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/chanvalue.c.o
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/clip.c.o
[ 12%] Linking C static library libreetz.a
[ 12%] Building C object src/rt/CMakeFiles/radiance.dir/m_clip.c.o
[ 12%] Building C object src/rt/CMakeFiles/radiance.dir/m_direct.c.o
[ 12%] Built target raycalls
[ 12%] Building C object src/rt/CMakeFiles/radiance.dir/m_mirror.c.o
[ 12%] Built target reetz
[ 12%] Building C object src/rt/CMakeFiles/radiance.dir/m_mist.c.o
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/color.c.o
[ 12%] Building C object src/common/CMakeFiles/rtrad.dir/colrops.c.o
[ 12%] Building C object src/rt/CMakeFiles/radiance.dir/mx_data.c.o
[ 13%] Building C object src/rt/CMakeFiles/radiance.dir/mx_func.c.o
[ 14%] Building C object src/common/CMakeFiles/rtrad.dir/cone.c.o
[ 14%] Building C object src/rt/CMakeFiles/radiance.dir/noise3.c.o
[ 14%] Building C object src/rt/CMakeFiles/radiance.dir/normal.c.o
[ 14%] Building C object src/rt/CMakeFiles/radiance.dir/o_cone.c.o
[ 14%] Building C object src/common/CMakeFiles/rtrad.dir/cvtcmd.c.o
[ 14%] Building C object src/rt/CMakeFiles/radiance.dir/o_face.c.o
[ 14%] Linking C static library librgl.a
[ 14%] Building C object src/common/CMakeFiles/rtrad.dir/depthcodec.c.o
[ 14%] Building C object src/common/CMakeFiles/rtrad.dir/dircode.c.o
[ 14%] Building C object src/rt/CMakeFiles/radiance.dir/o_instance.c.o
[ 14%] Built target rgl
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/o_mesh.c.o
[ 15%] Building C object src/common/CMakeFiles/rtrad.dir/disk2square.c.o
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/p_data.c.o
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/p_func.c.o
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/pmap.c.o
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/pmapamb.c.o
[ 15%] Building C object src/rt/CMakeFiles/radiance.dir/pmapbias.c.o
[ 16%] Building C object src/rt/CMakeFiles/radiance.dir/pmapcontrib.c.o
[ 16%] Building C object src/common/CMakeFiles/rtrad.dir/ealloc.c.o
[ 16%] Building C object src/rt/CMakeFiles/radiance.dir/pmapdata.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/eputs.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/erf.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/error.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/expandarg.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/face.c.o
[ 17%] Building C object src/rt/CMakeFiles/radiance.dir/pmapdiag.c.o
[ 17%] Building C object src/common/CMakeFiles/rtrad.dir/ezxml.c.o
[ 17%] Building C object src/rt/CMakeFiles/radiance.dir/pmapio.c.o
[ 17%] Building C object src/rt/CMakeFiles/radiance.dir/pmapmat.c.o
[ 17%] Building C object src/rt/CMakeFiles/radiance.dir/pmapopt.c.o
[ 18%] Building C object src/rt/CMakeFiles/radiance.dir/pmapparm.c.o
[ 18%] Building C object src/common/CMakeFiles/rtrad.dir/falsecolor.c.o
[ 18%] Building C object src/common/CMakeFiles/rtrad.dir/fdate.c.o
[ 18%] Building C object src/common/CMakeFiles/rtrad.dir/fgetline.c.o
[ 19%] Building C object src/rt/CMakeFiles/radiance.dir/pmaprand.c.o
[ 19%] Building C object src/common/CMakeFiles/rtrad.dir/fgetval.c.o
[ 19%] Building C object src/rt/CMakeFiles/radiance.dir/pmapray.c.o
[ 19%] Building C object src/common/CMakeFiles/rtrad.dir/fgetword.c.o
[ 19%] Building C object src/common/CMakeFiles/rtrad.dir/fixargv0.c.o
In file included from /build/source/src/common/falsecolor.c:16:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/fltdepth.c.o
[ 20%] Linking C static library libmgf.a
[ 20%] Building C object src/rt/CMakeFiles/radiance.dir/pmapsrc.c.o
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/fputword.c.o
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/font.c.o
[ 20%] Building C object src/rt/CMakeFiles/radiance.dir/pmaptype.c.o
[ 20%] Building C object src/rt/CMakeFiles/radiance.dir/pmcontrib2.c.o
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/free_os.c.o
[ 20%] Built target mgf
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/fropen.c.o
[ 20%] Building C object src/rt/CMakeFiles/radiance.dir/pmutil.c.o
[ 20%] Building C object src/common/CMakeFiles/rtrad.dir/fvect.c.o
[ 21%] Building C object src/common/CMakeFiles/rtrad.dir/gethomedir.c.o
[ 21%] Building C object src/rt/CMakeFiles/radiance.dir/preload.c.o
[ 21%] Building C object src/rt/CMakeFiles/radiance.dir/raytrace.c.o
[ 21%] Building C object src/rt/CMakeFiles/radiance.dir/renderopts.c.o
[ 22%] Building C object src/common/CMakeFiles/rtrad.dir/getlibpath.c.o
[ 22%] Building C object src/common/CMakeFiles/rtrad.dir/getpath.c.o
[ 22%] Building C object src/common/CMakeFiles/rtrad.dir/header.c.o
[ 22%] Building C object src/common/CMakeFiles/rtrad.dir/hilbert.c.o
[ 22%] Building C object src/rt/CMakeFiles/radiance.dir/source.c.o
[ 22%] Building C object src/rt/CMakeFiles/radiance.dir/sphere.c.o
[ 22%] Building C object src/common/CMakeFiles/rtrad.dir/image.c.o
[ 22%] Building C object src/rt/CMakeFiles/radiance.dir/srcobstr.c.o
[ 23%] Building C object src/rt/CMakeFiles/radiance.dir/srcsamp.c.o
[ 23%] Building C object src/common/CMakeFiles/rtrad.dir/idmap.c.o
[ 23%] Building C object src/rt/CMakeFiles/radiance.dir/srcsupp.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/instance.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/interp2d.c.o
[ 24%] Building C object src/rt/CMakeFiles/radiance.dir/t_data.c.o
[ 24%] Building C object src/rt/CMakeFiles/radiance.dir/t_func.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/invmat4.c.o
[ 24%] Building C object src/rt/CMakeFiles/radiance.dir/text.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/jitteraperture.c.o
[ 24%] Building C object src/rt/CMakeFiles/radiance.dir/virtuals.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/lamps.c.o
[ 24%] Building C object src/common/CMakeFiles/rtrad.dir/linregr.c.o
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/loadbsdf.c.o
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/loadvars.c.o
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/lookup.c.o
[ 25%] Linking C static library libwfobj.a
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/mat4.c.o
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/mesh.c.o
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/modobject.c.o
/build/source/src/rt/text.c:80:15: warning: conflicting types for built-in function 'gettext'; expected 'char *(const char *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wbuiltin-declaration-mismatch-Wbuiltin-declaration-mismatch8;;]
   80 | static TEXT * gettext(OBJREC *tm);
      |               ^~~~~~~
[ 25%] Building C object src/common/CMakeFiles/rtrad.dir/myhostname.c.o
[ 26%] Building C object src/common/CMakeFiles/rtrad.dir/multisamp.c.o
[ 26%] Building C object src/common/CMakeFiles/rtrad.dir/normcodec.c.o
[ 26%] Built target wfobj
[ 26%] Building C object src/common/CMakeFiles/rtrad.dir/objset.c.o
[ 26%] Building C object src/common/CMakeFiles/rtrad.dir/octree.c.o
[ 26%] Building C object src/common/CMakeFiles/rtrad.dir/otypes.c.o
[ 27%] Building C object src/common/CMakeFiles/rtrad.dir/paths.c.o
[ 27%] Building C object src/common/CMakeFiles/rtrad.dir/plocate.c.o
[ 27%] Building C object src/common/CMakeFiles/rtrad.dir/portio.c.o
[ 27%] Building C object src/common/CMakeFiles/rtrad.dir/process.c.o
[ 27%] Building C object src/common/CMakeFiles/rtrad.dir/quit.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/readfargs.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/readmesh.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/readobj.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/readoct.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/resolu.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/rexpr.c.o
[ 28%] Building C object src/common/CMakeFiles/rtrad.dir/savestr.c.o
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/savqstr.c.o
/build/source/src/common/readmesh.c: In function 'readmesh':
/build/source/src/common/readmesh.c:257:9: warning: 'checkheader' accessing 64 bytes in a region of size 15 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  257 |         checkheader(meshfp, MESHFMT, flags&IO_INFO ? stdout : (FILE *)NULL);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/common/readmesh.c:257:9: note: referencing argument 2 of type 'char[64]'
In file included from /build/source/src/common/standard.h:10,
                 from /build/source/src/common/readmesh.c:11:
/build/source/src/common/rtio.h:64:17: note: in a call to function 'checkheader'
   64 | extern int      checkheader(FILE *fin, char fmt[MAXFMTLEN], FILE *fout);
      |                 ^~~~~~~~~~~
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/sceneio.c.o
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/spec_rgb.c.o
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/tcos.c.o
/build/source/src/common/readobj.c: In function 'readobj':
/build/source/src/common/readobj.c:57:25: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Wunused-result8;;]
   57 |                         fgets(buf, sizeof(buf), infp);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/timegm.c.o
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/tmap16bit.c.o
/build/source/src/common/readoct.c: In function 'readoct':
/build/source/src/common/readoct.c:74:13: warning: 'checkheader' accessing 64 bytes in a region of size 16 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
   74 |         if (checkheader(infp, OCTFMT, load&IO_INFO ? stdout : (FILE *)NULL) < 0)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/common/readoct.c:74:13: note: referencing argument 2 of type 'char[64]'
In file included from /build/source/src/common/standard.h:10,
                 from /build/source/src/common/readoct.c:15:
/build/source/src/common/rtio.h:64:17: note: in a call to function 'checkheader'
   64 | extern int      checkheader(FILE *fin, char fmt[MAXFMTLEN], FILE *fout);
      |                 ^~~~~~~~~~~
[ 29%] Building C object src/common/CMakeFiles/rtrad.dir/tmapluv.c.o
[ 30%] Building C object src/common/CMakeFiles/rtrad.dir/tmapcolrs.c.o
[ 30%] Building C object src/common/CMakeFiles/rtrad.dir/tmaptiff.c.o
[ 30%] Building C object src/common/CMakeFiles/rtrad.dir/tmesh.c.o
[ 30%] Building C object src/common/CMakeFiles/rtrad.dir/tonemap.c.o
[ 30%] Building C object src/common/CMakeFiles/rtrad.dir/triangulate.c.o
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/urand.c.o
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/urind.c.o
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/words.c.o
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/wordfile.c.o
In file included from /build/source/src/common/tmap16bit.c:14:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/wputs.c.o
In file included from /build/source/src/common/tmaptiff.c:15:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
In file included from /build/source/src/common/tmapluv.c:19:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
In file included from /build/source/src/common/tmapcolrs.c:18:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
[ 31%] Building C object src/common/CMakeFiles/rtrad.dir/xf.c.o
[ 32%] Building C object src/common/CMakeFiles/rtrad.dir/zeroes.c.o
In file included from /build/source/src/common/tonemap.c:18:
/build/source/src/common/tmprivat.h:59:1: warning: variably modified 'tmMesofact' at file scope
   59 | extern uby8     tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~~~
[ 32%] Building C object src/common/CMakeFiles/rtrad.dir/unix_process.c.o
/build/source/src/common/tonemap.c:666:1: warning: variably modified 'tmMesofact' at file scope
  666 | uby8    tmMesofact[BMESUPPER-BMESLOWER];
      | ^~~~
[ 32%] Building C object src/common/CMakeFiles/rtrad.dir/strlcpy.c.o
/build/source/src/common/readmesh.c:288:42: warning: argument 1 value '18446744073709551615' exceeds maximum object size 9223372036854775807 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Walloc-size-larger-than=-Walloc-size-larger-than=8;;]
  288 |                 mp->patch = (MESHPATCH *)calloc(mp->npatches,
      |                                          ^~~~~~~~~~~~~~~~~~~~
  289 |                                         sizeof(MESHPATCH));
      |                                         ~~~~~~~~~~~~~~~~~~
In file included from /build/source/src/common/rtmisc.h:8,
                 from /build/source/src/common/standard.h:11:
/nix/store/ij144ma6vs8acil8r9hgr8xkb1dp9azg-glibc-2.39-5-dev/include/stdlib.h:675:14: note: in a call to allocation function 'calloc' declared here
  675 | extern void *calloc (size_t __nmemb, size_t __size)
      |              ^~~~~~
[ 33%] Linking C static library libradiance.a
[ 33%] Built target radiance
[ 33%] Linking C static library librtrad.a
[ 33%] Built target rtrad
make: *** [Makefile:166: all] Error 2
error: boost::bad_format_string: format-string is ill-formed
```

</details>